internal/jobs/launch_instance_aws.go

```diff
@@ -60,33 +62,32 @@ func HandleLaunchInstanceAWS(ctx context.Context, job *worker.Job) {
 
 	logger = ptr.To(logger.With().Int64("reservation_id", args.ReservationID).Logger())
 	ctx = logger.WithContext(ctx)
-	nc := notifications.GetNotificationClient(ctx)
 
 	jobErr := DoEnsurePubkeyOnAWS(ctx, &args)
 	if jobErr != nil {
 		finishWithError(ctx, args.ReservationID, jobErr)
-		nc.FailedLaunch(ctx, args.ReservationID, jobErr)
 		return
 	}
 
 	jobErr = DoLaunchInstanceAWS(ctx, &args)
 	if jobErr != nil {
 		finishWithError(ctx, args.ReservationID, jobErr)
-		nc.FailedLaunch(ctx, args.ReservationID, jobErr)
 		return
 	}
+
 	jobErr = FetchInstancesDescriptionAWS(ctx, &args)
 	if jobErr != nil {
-		nc.FailedLaunch(ctx, args.ReservationID, jobErr)
-	} else {
-		nc.SuccessfulLaunch(ctx, args.ReservationID)
+		finishWithError(ctx, args.ReservationID, jobErr)
 	}
```

Comment:
This is the actual bug - instead of calling `finishWithError` we just sent a notification and skipped on the error.

Reply:
hmmm when thinking about it, shall we fail here? as this step is not required... and we will report error on it, Users might get confused :thinking: